comp/otelcol/otlp/components/processor/infraattributesprocessor/metrics_test.go

```diff
@@ -128,12 +127,13 @@ func TestInfraAttributesMetricProcessor(t *testing.T) {
 				Metrics:     MetricInfraAttributes{},
 				Cardinality: types.LowCardinality,
 			}
-			fakeTagger := taggerimpl.SetupFakeTagger(t)
-			defer fakeTagger.ResetTagger()
-			fakeTagger.SetTags("container_id://test", "test", []string{"container:id"}, nil, nil, nil)
-			fakeTagger.SetTags("deployment://namespace/deployment", "test", []string{"deployment:name"}, nil, nil, nil)
-			fakeTagger.SetTags(collectors.GlobalEntityID, "test", []string{"global:tag"}, nil, nil, nil)
-			factory := NewFactory(fakeTagger)
+			tc := &testTaggerClient{
+				m: make(map[string][]string),
+			}
```

Comment:
Could you add a constructor in `taggerclient_test.go` to simplify instantiating this struct throughout the test files?

Reply:
done, great idea thanks!